File: basic.py
def fractran(program_str):
    # Parse "3/2, 1/3" → [(3,2), (1,3)]
    fractions = []
    for part in program_str.split(','):
        part = part.strip()
        if '/' in part:
            num, den = map(int, part.split('/'))
            if num > 0 and den > 0:
                fractions.append((num, den))

    result = []
    n = 2

    MAX_STEPS = 10000

    for _ in range(MAX_STEPS):
        if len(result) >= 10:
            break
        result.append(n)

        updated = False
        for num, den in fractions:
            if n % den == 0:
                n = (n // den) * num
                updated = True
                break

        if not updated:
            break

    return result


# Tests
print(fractran("3/2, 1/3"))                     # [2, 3, 1]
print(fractran("3/2, 5/3, 1/5"))                # [2, 3, 5, 1]
print(fractran("3/2, 6/3"))                     # [2, 3, 6, 9, 18, 27, 54, 81, 162, 243]
print(fractran("2/7, 7/2"))                     # [2, 7, 2, 7, 2, 7, 2, 7, 2, 7]
print(fractran("17/91, 78/85, 19/51, 23/38, 29/33, 77/29, 95/23, 77/19, 1/17, 11/13, 13/11, 15/14, 15/2, 55/1"))
# → [2, 15, 825, 725, 1925, 2275, 425, 390, 330, 290]

File: basic.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Fractran {

    public static int[] fractran(String program) {
        // Parse fractions
        List<int[]> fractions = new ArrayList<>();
        for (String part : program.split(",")) {
            part = part.trim();
            if (part.contains("/")) {
                String[] nums = part.split("/");
                int num = Integer.parseInt(nums[0].trim());
                int den = Integer.parseInt(nums[1].trim());
                if (num > 0 && den > 0) {
                    fractions.add(new int[]{num, den});
                }
            }
        }

        List<Integer> result = new ArrayList<>();
        long n = 2; // use long to avoid overflow in larger sequences

        final int MAX_STEPS = 10000;

        for (int step = 0; step < MAX_STEPS; step++) {
            if (result.size() >= 10) break;
            result.add((int) n);

            boolean updated = false;
            for (int[] frac : fractions) {
                int num = frac[0];
                int den = frac[1];
                if (n % den == 0) {
                    n = (n / den) * num;
                    updated = true;
                    break;
                }
            }

            if (!updated) {
                break;
            }
        }

        // Convert to int array
        int[] arr = new int[result.size()];
        for (int i = 0; i < result.size(); i++) {
            arr[i] = result.get(i);
        }
        return arr;
    }

    public static void main(String[] args) {
        System.out.println(Arrays.toString(fractran("3/2, 1/3")));
        System.out.println(Arrays.toString(fractran("3/2, 5/3, 1/5")));
        System.out.println(Arrays.toString(fractran("3/2, 6/3")));
        System.out.println(Arrays.toString(fractran("2/7, 7/2")));
        System.out.println(Arrays.toString(fractran("17/91, 78/85, 19/51, 23/38, 29/33, 77/29, 95/23, 77/19, 1/17, 11/13, 13/11, 15/14, 15/2, 55/1")));
    }
}
